CardanoSharp.Wallet/Encoding/Bech32Decoder.cs

```diff
@@ -0,0 +1,99 @@
+﻿using CardanoSharp.Wallet.Extensions;
+using System;
+using System.Collections.Generic;
+
+namespace CardanoSharp.Wallet.Encoding
+{
+    public static partial class Bech32
+    {
+        /// <summary>
+        /// Converts a bech-32 encoded string back to its byte array representation.
+        /// </summary>
+        /// <exception cref="FormatException"/>
+        /// <param name="bech32EncodedString">Bech-32 encoded string.</param>
+        /// <param name="witVer">Witness version</param>
+        /// <param name="hrp">Human readable part</param>
+        /// <returns>Byte array of the given string.</returns>
+        public static byte[] Decode(string bech32EncodedString, out byte witVer, out string hrp)
+        {
+            byte[] b32Arr = Bech32Decode(bech32EncodedString, out hrp);
+            if (b32Arr.Length < CheckSumSize)
+            {
+                throw new FormatException("Invalid data length.");
+            }
+            if (!VerifyChecksum(hrp, b32Arr))
+            {
+                throw new FormatException("Invalid checksum.");
+            }
```

Comment:
We should really make things like this into a Try/Parse pattern. 
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/exceptions-and-performance